src/main/java/racingcar/domain/Cars.java

```diff
@@ -1,60 +1,70 @@
 package racingcar.domain;
 
-import racingcar.RandomNumberGenerator;
-
-import java.util.Collections;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Set;
+import java.util.*;
 import java.util.stream.Collectors;
 
 public class Cars {
-
     private static final int MAX_SIZE = 20;
+    private static final String NAME_DELIMITER = ",";
+
     private final List<Car> cars;
 
-    public Cars(final List<String> cars) {
+    public Cars(final List<Car> cars) {
         validateCars(cars);
-        validateSize(cars);
-        this.cars = cars.stream()
+        this.cars = cars;
+    }
+
+    public static Cars from (final List<Name> names) {
+        List<Car> cars = names.stream()
                 .map(Car::new)
                 .collect(Collectors.toList());
+        return new Cars(cars);
     }
 
-    private void validateCars(final List<String> cars) {
-
-        validateDuplicatedName(cars);
-        validateEmpty(cars);
+    public static Cars fromNameValues(final List<String> nameValues) {
+        List<Name> names = nameValues.stream()
+                .map(Name::new)
+                .collect(Collectors.toList());
+        return from(names);
     }
 
-    private void validateEmpty(final List<String> cars) {
-        if (cars.isEmpty()) {
-            throw new IllegalArgumentException("최소 하나 이상의 Car 객체가 존재해야합니다.");
-        }
+    private void validateCars(final List<Car> cars) {
```

Comment:
Cars 객체이므로 메서드명에서 생략해도 될 것 같아요!

```suggestion
    private void validate(final List<Car> cars) {
```

Reply:
그렇네요...!! 반영했습니다 👍